### PR TITLE
Add `store.Slab`

### PR DESCRIPTION
**Pull request by peterjan**
```
BenchmarkSlabs/Slab
BenchmarkSlabs/Slab-10                       100           1894078 ns/op        22144.31 MB/s
BenchmarkSlabs/PinnedSlab
BenchmarkSlabs/PinnedSlab-10                 100            828057 ns/op        50652.38 MB/s
```

### CODE REVIEW

File: persist/postgres/slabs_test.go [hunk 1]
```diff
@@ -29,6 +29,70 @@ func TestSlab(t *testing.T) {
...
+	for _, host := range hosts {
+		root := frand.Entropy256()
+		params.Sectors = append(params.Sectors, slabs.SectorPinParams{
+			Root:    root,
+			HostKey: host,
+		})
+		expectedSectors = append(expectedSectors, slabs.Sector{
+			Root:       root,
+			HostKey:    &host,
+			ContractID: nil, // not pinned to a contract
```
Comment (ChrisSchinnerl) [inline persist/postgres/slabs_test.go:48]: Considering that the only difference between a pinned slab and a slab is that the slab can have a contract id and a pinnedAt timestamp I'd make sure the test covers those edge cases as well 😅 For the latter by at least checking it's not zero.

File: persist/postgres/sectors_test.go [hunk 2]
```diff
@@ -1130,6 +1130,22 @@ func BenchmarkSlabs(b *testing.B) {
 		}
 	})
 
+	b.Run("PinnedSlab", func(b *testing.B) {
```
Comment (ChrisSchinnerl) [inline persist/postgres/sectors_test.go:1133]: Considering that a pinned slab is a subset of a slab right now, how about making a pinned slab a conversion method on the slab rather than a separate database method? Sure, a pinned slab requires one less join but the query is pretty fast.

cc @n8maninger

Comment (peterjan) [reply persist/postgres/sectors_test.go:1133]: I considered that actually but I don't like conversion methods in general.

edit: just remembered there's also `slabs.ErrUnrecoverable` which `PinnedSlab` returns if it's unrecoverable, so if we want to keep that the method would have to potentially return an error

edit2: avoiding the left joins is probably worth the trouble of adding a separate method on the store imo

**Review by ChrisSchinnerl — APPROVED** [review]

**Review by n8mgr — APPROVED** [review]
